### PR TITLE
ref(escalating): Remove forecast floor bound

### LINKED ISSUE #51947: Remove forecast floor bound in escalating algorithm
### Problem Statement

The forecast floor bound of 200 is not a good estimation of when an issue with < 30 events/hr is escalating.

### Solution Brainstorm

Remove the forecast floor bound of 200.
Change the forecast fetch fallback to call the forecast generation algorithm and log an exception instead of returning the default floor bound.

### Product Area

Issues

### PR DESCRIPTION
**Pull request by npusarla**
Remove floor bound for escalating issues algorithm.
Log exception and call forecast generation function on fetch if the forecast does not exist

fixes https://github.com/getsentry/sentry/issues/51947

### CODE REVIEW

File: src/sentry/issues/escalating_issues_alg.py [hunk 1]
```diff
@@ -78,17 +78,6 @@ def generate_issue_forecast(
 
     ts_max = max(ts_data)
 
-    # minimum ceiling for issues that don't have much data from the past week
-    floor_bound = 200
```
Comment (trillville) [inline src/sentry/issues/escalating_issues_alg.py:82]: should this be set to 0 instead of removed?

Comment (npusarla) [reply src/sentry/issues/escalating_issues_alg.py:82]: why set to 0?

Comment (trillville) [reply src/sentry/issues/escalating_issues_alg.py:82]: oh nevermind, ignore me

**Review by trillville — COMMENTED** [review]
LGTM - i think you need to update a few of the unit tests btw

File: src/sentry/issues/escalating_group_forecast.py [hunk 2]
```diff
@@ -48,21 +47,29 @@ def save(self) -> None:
...
-        return EscalatingGroupForecast(
-            project_id=project_id,
-            group_id=group_id,
-            forecast=DEFAULT_MINIMUM_CEILING_FORECAST,
-            date_added=datetime.now(),
+        logger.exception(
+            "Forecast does not exist for project id: "
+            + str(project_id)
+            + " group id: "
+            + str(group_id)
```
Comment (snigdhas) [inline src/sentry/issues/escalating_group_forecast.py:60]:
```suggestion
        logger.exception(
            f"Forecast does not exist for project id: {str(project_id)} group id: {str(group_id)}"
        )
```

File: tests/sentry/issues/test_ignored.py [hunk 3]
```diff
@@ -84,10 +86,14 @@ def test_archive_until_escalating_no_counts(
...
-        assert fetched_forecast is not None
-        assert fetched_forecast.project_id == self.group.project.id
-        assert fetched_forecast.group_id == self.group.id
-        assert fetched_forecast.forecast == DEFAULT_MINIMUM_CEILING_FORECAST
+        assert fetched_forecast is None
+        assert mock_logger.exception.call_args.args[0] == (
+            "Forecast does not exist for project id: "
+            + str(self.group.project.id)
+            + " group id: "
+            + str(self.group.id)
```
Comment (snigdhas) [inline tests/sentry/issues/test_ignored.py:94]: same as above, f-strings are a bit easier to read IMO

File: tests/sentry/tasks/test_weekly_escalating_forecast.py [hunk 4]
```diff
@@ -27,25 +24,31 @@ def create_archived_until_escalating_groups(self, num_groups: int) -> List[Group
...
-            fetched_forecast = EscalatingGroupForecast.fetch(
-                group_list[0].project.id, group_list[0].id
+            group = group_list[0]
+            fetched_forecast = EscalatingGroupForecast.fetch(group.project.id, group.id)
+            assert fetched_forecast is None
+            assert mock_logger.exception.call_args.args[0] == (
+                "Forecast does not exist for project id: "
+                + str(group.project.id)
+                + " group id: "
+                + str(group.id)
```
Comment (snigdhas) [inline tests/sentry/tasks/test_weekly_escalating_forecast.py:49]: same thing here

**Review by snigdhas — APPROVED** [review]
a few nits, but LGTM!

**Pull request merged**